misk/src/main/kotlin/misk/web/actions/UpstreamResourceInterceptor.kt

```diff
@@ -0,0 +1,69 @@
+package misk.web.actions
+
+import misk.web.NetworkChain
+import misk.web.NetworkInterceptor
+import misk.web.Response
+import misk.web.toMisk
+import misk.web.toOkHttp3
+import okhttp3.HttpUrl
+
+class UpstreamResourceInterceptor(
+  private val mappings: MutableList<out Mapping>
+) : NetworkInterceptor {
+  @Suppress("UNUSED_PARAMETER")
+  override fun intercept(chain: NetworkChain): Response<*> {
+    val requestSegments = chain.request.url.pathSegments()
+    var matchedMapping: Mapping? = null
+
+    for (mapping in this.mappings) {
+      if (!pathSegmentsMatch(mapping.localPathPrefix.drop(1).dropLast(1).split('/'), requestSegments)) continue
+      if (matchedMapping == null || mapping.localPathPrefix.count { ch -> ch == '/' } > matchedMapping.localPathPrefix.count { ch -> ch == '/' }) matchedMapping = mapping
+    }
+
+    if (matchedMapping == null) return chain.proceed(chain.request)
+
+    val upstreamUrl = matchedMapping.upstreamBaseUrl
+    val upstreamSegments = upstreamUrl.pathSegments()
+    val upstreamPlusRequestSegments = upstreamSegments.subList(0, upstreamSegments.size - 1) + requestSegments.subList(
```

Comment:
I think this will also be easier as a string replacement. Just take the suffix of the request after the localPathPrefix and append that to the upstreamUrl

Reply:
This too 👍